### DISCUSSION TITLE
can we add support for paging？

### DISCUSSION THREAD

**Discussion by zey1996** [grafana/clickhouse-datasource#449] [Ideas]
When I was querying a lot of logs. The performance of the grafana was not very good.
Perhaps we can add a paging feature to make it easier to query large amounts of logs or data?

**Comment by asimpson** [comment]
Thanks for reaching out! Can you give more details on the queries that slowed Grafana down, e.g. how many logs returned, example query etc?

**Reply by zey1996** [reply]
When I query 10,000 logs, it's very slow, but it's still can work. but when i query log limit 100000, it's carshed. 
The query like `SELECT * FROM database.table WHERE   ( timestamp>= $__fromTime AND timestamp<= $__toTime ) LIMIT 100000`
I think clickhouse is ok, but browser can't do it.
I use chrome, and I have more than 10G free memory for chrome.
I think it's a question of grafana, but if we can support paging, it;s may be better.

**Comment by SpencerTorres** [comment]
This would need to be done on the Grafana panel/UI, It wouldn't make much sense to have it within the query builder. We already have the `LIMIT` for limiting data.

I also tried reading 10,000 logs and the only issue I had was the HTML loading in as I scrolled. Seems like it wasn't that much data in memory, just a lot to render
